cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -79,13 +87,53 @@
   @ClassRule
   public static final TemporaryFolder TEMP_FOLDER = new TemporaryFolder();
 
+  private static NamespaceId configuredNamespace = configureTestNamespace();
+  // boolean is true if we created the namespace; indicates that we are responsible for cleaning it
+  private static Map<NamespaceId, Boolean> registeredNamespaces = new HashMap<>();
+
   private AccessToken accessToken;
 
   @Before
   public void setUp() throws Exception {
     checkSystemServices();
     assertUnrecoverableResetEnabled();
-    assertIsClear();
+
+    boolean createdByUs = false;
+    if (!getNamespaceClient().exists(configuredNamespace.toId())) {
```

Comment:
correct; just with the idea, that if we didn't create it, we're not responsible for deleting it.
That way, the impersonation doesn't have to be set up again on the next run.